src/bidiMapper/domains/script/RealmStorage.ts

```diff
@@ -62,13 +62,21 @@ export class RealmStorage {
       }
       if (
         filter.browsingContextId !== undefined &&
-        filter.browsingContextId !== realm.browsingContextId
+        (!(realm instanceof WindowRealm) ||
+          filter.browsingContextId !== realm.browsingContext.id)
```

Comment:
`realm.associatedBrowsingContexts.includes(filter.browsingContextId)`?

Reply:
`associatedBrowsingContexts` are not necessarily the owners of the given realm. The only realm that has a browsing context is `WindowRealm`. Every other realm is indirectly related. Let me rename this to better imply this.

Reply:
I think the model of `associatedBrowsingContexts` is perfectly fine, as you can request any realm associated with the given BrowsingContext. In filter you can specify a `type`, if you want a `window` realm.

Reply:
There can other from `window` realms associated with the given browsing context, and I would expect the filter `{browsingContextId: SOME_VALUE}` to return all of them.

Reply:
Eg BiDi command [`script.getRealms`](https://w3c.github.io/webdriver-bidi/#command-script-getRealms) should return not only window realms, but workers as well.

Reply:
That warrants it. Changed.